common/auth/tls_config_helper.go

```diff
@@ -125,86 +120,130 @@ func NewTLSConfig(temporalTls *TLS) (*tls.Config, error) {
 	}
 
 	// Load CA cert
+	caCertPool, err := parseCAs(temporalTls)
+	if err != nil {
+		return nil, err
+	}
+	if caCertPool != nil {
+		tlsConfig.RootCAs = caCertPool
+	}
+
+	// Load client cert
+	clientCert, err := parseClientCert(temporalTls)
+	if err != nil {
+		return nil, err
+	}
+	if clientCert != nil {
+		tlsConfig.Certificates = []tls.Certificate{*clientCert}
+	}
+
+	return tlsConfig, nil
+}
+
+func validateTemporalTls(temporalTls *TLS) error {
+	if temporalTls.CertData != "" && temporalTls.CertFile != "" {
+		return fmt.Errorf("%w: %s", ErrTLSConfig, "only one of certData or certFile properties should be specified")
+	}
+
+	if temporalTls.KeyData != "" && temporalTls.KeyFile != "" {
+		return fmt.Errorf("%w: %s", ErrTLSConfig, "only one of keyData or keyFile properties should be specified")
+	}
+
+	certProvided := temporalTls.CertData != "" || temporalTls.CertFile != ""
+	keyProvided := temporalTls.KeyData != "" || temporalTls.KeyFile != ""
+	if certProvided && !keyProvided {
```

Comment:
What if `certProvided = false` and `keyProvided = true`?
Isn't it also an error?
Isn't the correct condition `certProvided != keyProvided` and change the error message saying that both must be provided?